File: app/src/main/java/com/mandarin/bcu/androidutil/fakeandroid/FTMT.java
package com.mandarin.bcu.androidutil.fakeandroid;

import android.graphics.Matrix;

import common.system.fake.FakeTransform;

public class FTMT implements FakeTransform {
    private final Matrix m = new Matrix();

    FTMT(Matrix m) {
        this.m.set(m);
    }

    @Override
    public Object getAT() {
        return m;
    }

    void updateMatrix(Matrix m) {
        this.m.set(m);
    }

    public void setMatrix(Matrix m) {
        m.set(this.m);
    }
}

File: app/src/main/java/com/mandarin/bcu/androidutil/fakeandroid/FTMT.kt
package com.mandarin.bcu.androidutil.fakeandroid

import android.graphics.Matrix
import common.system.fake.FakeTransform

class FTMT internal constructor(m: Matrix?) : FakeTransform {
    private val m = Matrix()
    override fun getAT(): Any {
        return m
    }

    fun updateMatrix(m: Matrix?) {
        this.m.reset()
        this.m.set(m)
    }

    fun setMatrix(m: Matrix) {
        m.set(this.m)
    }

    init {
        this.m.set(m)
    }
}